libultraship/libultraship/ControlDeck.cpp

```diff
@@ -73,22 +74,39 @@ namespace Ship {
     }
 
 	void ControlDeck::WriteToPad(OSContPad* pad) const {
-		bool shouldBlockGameInput = CVar_GetS32("gOpenMenuBar", 0) && CVar_GetS32("gControlNav", 0);
+        // We block controller input if F1 menu is open and control navigation is on.
+        // This is because we don't want controller inputs to affect the game
+		bool shouldBlockControllerInputs = CVar_GetS32("gOpenMenuBar", 0) && CVar_GetS32("gControlNav", 0);
+
+        // We block keyboard input if you're currently typing into a textfield.
+        // This is because we don't want your keyboard typing to affect the game.
+        ImGuiIO io = ImGui::GetIO();
+        bool shouldBlockKeyboardInputs = io.WantCaptureKeyboard;
 
 		for (size_t i = 0; i < virtualDevices.size(); i++) {
 			const std::shared_ptr<Controller> backend = physicalDevices[virtualDevices[i]];
+
+            // If the controller backend is "Auto" we need to get the real device
+            // we search for the real device to read input from it
 			if (backend->GetGuid() == "Auto") {
 				for (const auto& device : physicalDevices) {
-					if(shouldBlockGameInput && device->GetGuid() != "Keyboard") {
-						continue;
-					}
+					if(shouldBlockControllerInputs && device->GetGuid() != "Keyboard")
+                        continue;
+
+                    if (shouldBlockKeyboardInputs && device->GetGuid() == "Keyboard")
+                        continue;
```

Comment:
i'm cool with
```c
if (shouldBlockKeyboardInputs && device->GetGuid() == "Keyboard") continue;
```
and
```c
if (shouldBlockKeyboardInputs && device->GetGuid() == "Keyboard") {
    continue;
}
```

but 2 line if statements are scary (a bad merge can break things, we don't want to have a [goto fail moment](https://www.wired.com/2014/02/gotofail/))